### PR TITLE
Implement online verification for darcs

### PR DESCRIPTION
**Pull request by kc1212**
This PR allows verification of darcs when the signature path is empty,
but there exists a digest of the signature path. The server, given an
appropriate callbacks to find darcs, is able to find the missing path
and use that for the verification.  This is only the first step, more
improvements are planned in the future, which we describe in #33.

### CODE REVIEW

File: omniledger/darc/darc.go [hunk 1]
```diff
@@ -414,20 +416,37 @@ func NewDarcSignature(signer *Signer, id ID, path []*Darc) (*Signature, error) {
 		return nil, errors.New("path missing")
 	}
 
-	hash, err := hashAll(id, darcsMsg(path))
+	// We create the message and then sign it, make sure the message is
```
Comment (ineiti) [inline omniledger/darc/darc.go]: s/We c/C/

And we should discuss whether it's worth to discuss in what person we should write comments. Is it `We`, `It`, `I`, ...? I always struggle and probably change it according to the time of day...

File: omniledger/darc/darc_test.go [hunk 2]
```diff
@@ -145,6 +145,71 @@ func TestDarc_EvolveMore(t *testing.T) {
...
+		expression.InitOrExpr(prevOwner.Identity().String())))
+
+	darcs := []*Darc{d}
+	for i := 0; i < 10; i++ {
+		dNew := darcs[len(darcs)-1].Copy()
+		dNew.IncrementVersion()
+		newOwner := NewSignerEd25519(nil, nil)
+		require.Nil(t, dNew.Rules.UpdateEvolution([]byte(newOwner.Identity().String())))
+		require.Nil(t, dNew.Evolve(darcs, prevOwner))
+		// require.Nil(t, dNew.Verify())
```
Comment (ineiti) [inline omniledger/darc/darc_test.go]: Why is that commented out?

**Comment by ineiti** [comment]
A late comment on `Rules.UpdateEvolution`: if you treat this rule special then you should also add `Rules.UpdateSigner`. Both `Evolution` and `Signer` (or whatever it will be called) are basic rules that have special meaning within darcs. So I would propose to either have both as methods on `Rules`, or none.

What do you think? I can also open an issue to discuss.

**Comment by kc1212** [comment]
Thanks @ineiti I created an issue for this https://github.com/dedis/student_18_omniledger/issues/36

File: omniledger/darc/darc.go (hunk 1 above)
Comment (kc1212) [reply omniledger/darc/darc.go]: I usually just use "we", that's kind of the default when writing papers at least

File: omniledger/darc/darc_test.go (hunk 2 above)
Comment (kc1212) [reply omniledger/darc/darc_test.go]: probably as a result of debugging, it's uncommented now

**Pull request merged**